### PR TITLE
Generate `.d.ts` files for all packages

### LINKED ISSUE #11722: [TS] Decide how to generate type definitions & enable testing them
Part of #11704. This is to achieve #504. See also #11720.

---

*   TODO:
    *   Agree what will we be publishing for each package (ckeditor5, ckeditor5-build-_, ckeditor5-_) on npm → Create an ADR/RFC → Verify with the entire CKE5 team.
    *   Write down how to simulate using the future packages with types.
    *   Decide on whether each package should contain its typings or we should bring them in a separate package (so other packages, like ckeditor5-react, may depend on the typings without depending on the entire CKEditor 5's codebase).
    *   Other random questions
        *   BTW: will this be `src/index.d.ts` only or multiple files?
        *   BTW: separate package with types? is that a common practice, if so, why?
        *   Check out how other similar projects publish their typings. Are there any good practices?
*   This task does not include updating the release process. This will be handled separately.
*   Some ideas: Publish to npm a beta-version? Or find a workflow on how to test it completely locally? E.g. verdaccio? or [https://github.com/wclr/yalc](https://github.com/wclr/yalc)
*   This task will unblock #13471.

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal: Generate `.d.ts` files for all packages. Closes #11722.

### CODE REVIEW

**Review by pomek — APPROVED** [review]
LGTM.

**Comment by pomek** [comment]
One thing, please add the `src/*.d.ts` entry in an array below:

https://github.com/ckeditor/ckeditor5/blob/446701b6e8862aeead7fc132e8611cc5ae91ec1a/scripts/release/publish.js#L43-L52

It defines what the `ckeditor5` package will contain when releasing the CKEditor 5 project.

**Comment by filipsobol** [comment]
Done :heavy\_check\_mark:

**Review by arkflpc — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -81,5 +81,7 @@ packages/ckeditor5-watchdog/src/**/*.js
 packages/ckeditor5-widget/src/**/*.js
 packages/ckeditor5-word-count/src/**/*.js
 packages/*/src/**/*.d.ts
+packages/*/src/**/*.js.map
```
Comment (pomek) [inline .gitignore:84]: Shouldn't we remove maps too? 

https://github.com/ckeditor/ckeditor5/blob/d88f9d31acbb4be6e3b10124c2c1ecf45bfed78a/scripts/release/clean.js#L39-L42

**Pull request merged**